src/sql/workbench/contrib/dataExplorer/browser/dataExplorer.contribution.ts

```diff
@@ -22,13 +23,23 @@ configurationRegistry.registerConfiguration({
 	'title': localize('databaseConnections', "Database Connections"),
 	'type': 'object',
 	'properties': {
-		'datasource.connections': {
-			'description': localize('datasource.connections', "data source connections"),
+		[CONNECTIONS_CONFIG_KEY]: {
+			'description': localize(CONNECTIONS_CONFIG_KEY, "data source connections"),
 			'type': 'array'
 		},
-		'datasource.connectionGroups': {
-			'description': localize('datasource.connectionGroups', "data source groups"),
+		[GROUPS_CONFIG_KEY]: {
+			'description': localize(GROUPS_CONFIG_KEY, "data source groups"),
 			'type': 'array'
+		},
+		[CONNECTIONS_SORT_BY_CONFIG_KEY]: {
+			'type': 'string',
+			'enum': [ConnectionsSortBy.dateAdded, ConnectionsSortBy.displayName],
+			'enumDescriptions': [
+				localize('connections.sortBy.dateAdded', 'Saved connections are sorted by the dates they were added.'),
+				localize('connections.sortBy.displayName', 'Saved connections are sorted by their display names alphabetically.')
+			],
+			'default': ConnectionsSortBy.dateAdded,
+			'description': localize(CONNECTIONS_SORT_BY_CONFIG_KEY, "Order used for sorting saved connections and connection groups")
```

Comment:
Let's just use a string constant for the ID - that doesn't need to actually match the config key name and so we don't want to tie them together since they aren't really related at all (the config key could change but we don't want to have that result in a new "string" needing to be localized)

Reply:
updated